### PR TITLE
Basic Singletask test-suite for ML models learning.

### PR DESCRIPTION
**Pull request by rbharath**
Will update once complete.

### CODE REVIEW

**Comment by rbharath** [comment]
Changes in this PR:

- Added `mode` to `Metric` objects, where `mode == "classification"` or `mode == "regression"`. 
- Added method `predict_proba()` and `predict_proba_on_batch()` to the `Model` class. This function follows the `sklearn` method `predict_proba` that predicts class probabilities for classifiers. The `predict_proba()` method produces output of shape `(n_samples, n_tasks, n_classes)`. 
- Added a simple learning benchmark. Verifies that the various `deepchem` models can overfit trivial single-task classification and regression benchmarks.
